src/strand-repeater-row/strand-repeater-row.js

```diff
@@ -0,0 +1,129 @@
+(function(scope) {
+	var DataUtils = StrandLib.DataUtils;
+
+	scope.RepeaterRow = Polymer({
+		is: 'strand-repeater-row',
+
+		behaviors: [
+			StrandTraits.LightDomGettable,
+			StrandTraits.Stylable
+		],
+
+		properties: {
+			model: {
+				type: Object,
+				notify: true,
+				observer: '_modelChanged'
+			},
+			scope: {
+				type: Object,
+				notify: true
+			},
+			index: {
+				type: Number
+			},
+
+			errors: {
+				type: Array,
+				value: false,
+				notify: true
+			},
+
+			_shouldShowAddRow: {
+				type: Boolean,
+				computed: '_computeShouldShowAddRow(index, scope._last, scope.maxRows)',
+				notify: true
+			}
+		},
+
+		observers: [
+			'_notifyModelChanged(model.*)'
+		],
+
+		_notifyModelChanged: function(changeRecord) {
+			this.scope._notifyModelChanged(changeRecord);
+		},
+
+		_addRow: function() {
+			this.scope.add();
+		},
+
+		_removeRow: function(ev) {
+			this.scope.remove(this.index);
+		},
+
+		_computeContainerClass: function(index, last) {
+			var o = {};
+			o["container"] = true;
+			o["has-bottom-margin"] = index !== last;
+			return this.classBlock(o);
+		},
+
+		_computeShouldShowAddRow: function(index, last, maxRows) {
+			return (index === last) && (maxRows ? last < maxRows-1 : true);
+		},
+
+		_expandErrorContainer: function(errors) {
+			if(errors) {
+				errors = errors.filter(function(err) {
+					return DataUtils.isDef(err) && DataUtils.isDef(err.message);
+				});
+
+				if(errors.length > 0) {
+					return this.styleBlock({
+						position: "static",
+						visibility: "hidden"
+					});
+				}
+			}
+		},
+
+		_computeErrorMessageStyle: function(record) {
+			if(record && record.elt instanceof HTMLElement) {
+				var current = record.elt,
+					prev = Polymer.dom(current).previousElementSibling,
+
+					rowRect = this.getBoundingClientRect(),
+					rect = current.getBoundingClientRect(),
+					prevRect = prev ? prev.getBoundingClientRect() : null,
+
+					width = rect.width,
+					position = rect.left - rowRect.left,
+					offset = prev ? (prevRect.right - rowRect.left) : 0,
+
+					s = {
+						left: (position-offset)+"px",
+						width: width+"px"
+					};
+
+				return this.styleBlock(s);
+			}
+		},
+
+		_validate: function(record) {
+			if(DataUtils.isDef(record.cId) && record.cId === this.model.cId) {
+				this.errors = Object.keys(record).map(function(key) {
+					if(key !== 'cId') {
+						var elt = Polymer.dom(this).querySelector('[name="'+key+'"]');
+						var message = record[key];
+
+						// Set error state if there is an error message
+						if(elt) elt.error = !!message;
+
+						return {
+							name: key,
+							message: message,
+							elt: elt
+						}
+					} else {
+						return null;
+					}
+				}.bind(this));
```

Comment:
`this` can be passed as the second argument to `[ ].map`, no need to `.bind(this)`